samples/web/CMakeLists.txt

```diff
@@ -0,0 +1,135 @@
+#    /filaweb.js         copied from REPO/samples/web
+#    /favicon.png        copied from REPO/samples/web
+#    /monkey/*.png       copied from REPO/assets/models
+#    /monkey/*.filamesh  built by filamesh
+#    /desert/*           built by cmgen
+#
+# ==================================================================================================
+
+function(copy_assets PATH NAME)
+    file(COPY ${PATH}/${NAME} DESTINATION ${PROJECT_BINARY_DIR}/public)
+    add_custom_target(${NAME} ALL DEPENDS ${NAME})
+    add_dependencies(filament ${NAME})
+endfunction()
+
+copy_assets("." filaweb.js)
+copy_assets("." favicon.png)
+copy_assets("../../assets/models" monkey)
+
+# Convert OBJ into a simple binary format.
+set(source_mesh "${CMAKE_CURRENT_SOURCE_DIR}/../../assets/models/monkey/monkey.obj")
+add_custom_command(
+    OUTPUT public/monkey/mesh.filamesh
+    COMMAND filamesh ${source_mesh} public/monkey/mesh.filamesh
+    MAIN_DEPENDENCY ${source_mesh}
+    DEPENDS filamesh)
+
+# Next build two cubemaps: (1) a single miplevel of a slightly large blurry PNG for the skybox, and
+# (2) all miplevels for an RGBM (which is PNG in disguise) for the IBL.
+
+# TODO: invoke cmgen only once and simply use RGBM everywhere. The only reason we're using PNG for
+# the skybox is to work around an RGBM coloration issue, probably something wrong in how we're
+# decoding the alpha channel for images in the web app.
+
+set(source_envmap "${CMAKE_CURRENT_SOURCE_DIR}/../../assets/environments/desert/desert.exr")
+add_custom_command(
+    OUTPUT public/desert/m0_nx.rgbm
+    COMMAND cmgen -x public --format=rgbm --size=256 ${source_envmap} 
+    MAIN_DEPENDENCY ${source_envmap}
+    DEPENDS cmgen)
+add_custom_command(
+    OUTPUT public/desert/nx.png
+    COMMAND cmgen -x public --format=png --size=512 --extract-blur=0.1 ${source_envmap} 
+    MAIN_DEPENDENCY ${source_envmap}
+    DEPENDS cmgen)
+
+# ==================================================================================================
+# For each demo, build a pair of wasm/js files and copy over an HTML file.
+# ==================================================================================================
+
+set(EXPORTS "'_launch','_main','_render','_resize'")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s EXPORTED_FUNCTIONS=[${EXPORTS}]")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s ALLOW_MEMORY_GROWTH=1")
+if (CMAKE_BUILD_TYPE MATCHES Debug)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s ASSERTIONS=2")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s SAFE_HEAP=1 ")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s DEMANGLE_SUPPORT=1")
+endif()
+
+function(add_demo NAME)
+    include_directories(${GENERATION_ROOT})
+    add_executable(${NAME} ${NAME}.cpp
+            filamesh.h
+            filamesh.cpp
+            filaweb.h
+            filaweb.cpp
+            public/monkey/mesh.filamesh
+            public/desert/m0_nx.rgbm
```

Comment:
Why do you grab only one cubemap face?

Reply:
This merely establishes a file-level dependency between the "executable" and the asset-building step.  Agreed that this need to be cleaned up.